Extend docs to include macOs options access

sambhavnoobcoder opened the pull request:
### Summary

This is a…

- [ ] Correction
- [x] Addition
- [ ] Restructuring
- [ ] Minor / simple change (like a typo)
- [ ] Other

### Reason for the Change

I was trying to address was how to access the plugins for Picard , which was discussed in the thread here : https://community.metabrainz.org/t/how-to-run-plugins-on-picard-as-a-developer/674175 .I think that clearly mentioning the access to plugins or options menu to MacOs users would be more intuitive and useful .

Jira Ticket _(optional)_ - [PICARD-2818](https://tickets.metabrainz.org/browse/PICARD-2818)

### Description of the Change

I essentially added the path of action which would lead to the user reaching the plugins selection via the options menu in a MacOs environment , extending the guide to the windows based systems . adjoining with the pr are screenshots demonstrating the working of the pathway updated in the pr . 

<img width="416" alt="Screenshot 2024-01-12 at 2 15 11 AM" src="https://github.com/metabrainz/picard-docs/assets/94298612/5b8393f7-7819-483f-a295-ee633b0a91bb">

img 1 . Python --> Preferences 

<img width="806" alt="Screenshot 2024-01-12 at 2 15 54 AM" src="https://github.com/metabrainz/picard-docs/assets/94298612/59a91eaf-487b-492c-8c04-46c4ac4f8431">

img 2. plugins selection 

### Additional Action Required

No Additional Action Required .

phw reviewed (changes requested):
Thanks. A few corrections.

phw commented on config/configuration.rst:
```diff
@@ -66,9 +66,10 @@ actions that Picard can perform when saving selected music files:
-a new window with the option groups listed in a tree format on the left hand side, and the individual
-settings on the right hand side.  This is where the majority of Picard's customization is performed.
+The option settings are found under the :menuselection:`"Options --> Options..."` item on the menu bar. Whereas in the MacOs ,the option
+settings are found under the :menuselection:`"Python --> Preferences ..."` item on the menu bar . This will open a new window with 
```
One thought: If we actually want to mention the variation when running from source it should be done as a foot note so it does not confuse normal users too much with development details.

rdswift replied:
In a `..  note::` block just below that paragraph / section, or as a proper footnote?  We use both within the documentation, so I think either one would be okay.

rdswift commented on config/configuration.rst:
```diff
@@ -66,9 +66,10 @@ actions that Picard can perform when saving selected music files:
-a new window with the option groups listed in a tree format on the left hand side, and the individual
-settings on the right hand side.  This is where the majority of Picard's customization is performed.
+The option settings are found under the :menuselection:`"Options --> Options..."` item on the menu bar. Whereas in the MacOs ,the option
+settings are found under the :menuselection:`"Python --> Preferences ..."` item on the menu bar . This will open a new window with 
```
I don't think there should be a space between "Preferences" and the three periods that follow.  At some point, I plan to replace all occurrences of three periods with a proper ellipsis, so you can do that here if you like (but not necessary).

sambhavnoobcoder commented:
okay , adding this as a note seperately does make more sense . I'll make that change and put up the pr .I'll also make changes to the macOs part as well in it along with the smaller space changes .

sambhavnoobcoder commented:
@phw , @rdswift I have put up a new commit addressing the issues in the review . The PR addresses the following issues :

- updated spelling of macOs to be consistent exactly like this throughout .
- punctuation is now `bar.` and not `bar .`.
- added the menu access information in a note just below the paragraph , instead of inside the paragraph using the `..  note::` block.
- removed the space between the 'Preferences' and the three dots . (I've kept the three periods for now , but I will gladly help @rdswift whenever he plans to replace them with the proper ellipsis throughout the document).

kindly review the new commit as well .

phw commented on config/configuration.rst:
```diff
@@ -66,9 +66,14 @@ actions that Picard can perform when saving selected music files:
-The option settings are found under the :menuselection:`"Options --> Options..."` item on the menu bar.  This will open
-a new window with the option groups listed in a tree format on the left hand side, and the individual
-settings on the right hand side.  This is where the majority of Picard's customization is performed.
+The option settings are found under the :menuselection:`"Options --> Options..."` item on the menu bar. This will open a new window with 
```
It should still mention the location for macOS users in general. Let's write e.g.

> The option settings are found under the :menuselection:`"Options --> Options..."` item on the menu bar. On macOS they can be accessed with :menuselection:`"MusicBrainz Picard --> Preferences...".

But please double check the exact spelling by running the Picard app (from https://picard.musicbrainz.org/downloads/). I don't have a mac environment available right now and don't remember whether the menu item shows up as "MusicBrainz Picard" or "Picard".

sambhavnoobcoder replied:
okay , I'll rerun it , make that change and put up the PR

sambhavnoobcoder commented:
@phw I have added the change to guide the user running the downloaded version of the application to the options menu . as for the "MusicBrainz Picard" v/s "Picard" issue , its kind both in Mac actually . when the application starts, as long as the application is in the phase where it displays the warning dialog box , it says Picard in the bar , however once the files / folder are uploaded the bar starts saying MusicBrainz Picard . I am attaching the images for both the cases below :

<img width="1440" alt="Screenshot 2024-01-12 at 10 10 01 PM" src="https://github.com/metabrainz/picard-docs/assets/94298612/3cf9c1c0-3169-4d64-b6d9-306440f71fc8">

img 1. illustration of the bar saying "Picard" when the app is first opened with the dialog box 

<img width="1440" alt="Screenshot 2024-01-12 at 10 14 28 PM" src="https://github.com/metabrainz/picard-docs/assets/94298612/efb00e6e-31c0-4ee8-a0d6-f808788b8c3f">

img 2 . illustration of the bar saying "MusicBrainz Picard" once files are uploaded .

My last commit adds the guide with reference to MusicBrainz , however should you want me to change that , I'll gladly comply .

rdswift commented on config/configuration.rst:
```diff
@@ -66,9 +66,15 @@ actions that Picard can perform when saving selected music files:
+format on the left hand side, and the individual settings on the right hand side. This is where the majority of Picard's 
+customization is performed.
+
+..  note:: 
```
This is a really minor item, but for consistency can you please remove one of the spaces between ".." and "note::"?  Thanks,

```suggestion
.. note:: 
```

I think the rest of the content is okay, but I've never used a mac in my life, so I can't speak to the way it handles the menu bar.

@sambhavnoobcoder, looking at your original images again it appears that the "Options" link appears to the right of "MusicBrainz Picard".  What happens if you click on this item?  Does it provide a sub-menu with an item to open the option settings window?  If so, what is the difference between this and using "MusicBrainz Picard --> Preferences..."?

phw replied:
> Does it provide a sub-menu with an item to open the option settings window?

No, on macOS that item is under the menu with the application name. Hence this clarification

sambhavnoobcoder commented:
@rdswift , the option link that you are seeing seems to be housing other features like move files , rename files , save tags , etc . I am attaching a screenshot of the option tab below :

<img width="823" alt="Screenshot 2024-01-15 at 4 49 18 PM" src="https://github.com/metabrainz/picard-docs/assets/94298612/e7be6342-b70f-4495-9e3d-a9a11ba44bb3">

img . options tab in macOS

I have also fixed the note whitespacing that you mentioned in the review above . kindly review that as well and merge if all is okay .

phw reviewed (approved):
Looks good, thanks a lot for both noticing the issue and the update.

phw commented:
CI reports some Sphinx incompatibility, but seems unrelated to this PR and rather is likely some issue with updated Python packages. Seems to be because of this recent update of https://pypi.org/project/sphinxcontrib-applehelp/1.0.8/ :

>The sphinxcontrib.applehelp extension used by this project needs at least Sphinx v5.0; it therefore cannot be built with this version.

We probably should follow and update to use Sphinx 5.0. Not sure what it involves.

This will not block us merging the PR, though. But I'll have @rdswift take a final look on the changes here.

phw commented:
> as for the "MusicBrainz Picard" v/s "Picard" issue , its kind both in Mac actually . when the application starts, as long as the application is in the phase where it displays the warning dialog box , it says Picard in the bar , however once the files / folder are uploaded the bar starts saying MusicBrainz Picard .

This is actually interesting. I'd count this as a bug in Picard, but unsure how and even if this can be resolved (Qt takes care of most of the handling for the global menu bar). But maybe application name is updated later on or something. It might also be because of the dialog itself. The global menu bar is a bit special, as it can change with dialogs.
